src/staff-administrator/components/staff-add-edit-user/staff-user-check-answers/staff-user-check-answers.component.ts

```diff
@@ -1,20 +1,211 @@
+      }
+    });
+  }
+
+  private prepareJobTitlesPayload() {
+    const jobTitlesNamePayload = [];
+    this.addUserData[9].value.map(jobTitle => {
+      jobTitlesNamePayload.push(this.staffFilterOptions.jobTitles.find(jobTitles => jobTitles.key === jobTitle));
+    });
+
+    jobTitlesNamePayload.map(jobTitleName => {
+      this.jobTitlesPayload.push(StaffJobTitles.jobTitles.find(jobTitle => jobTitle.role === jobTitleName.label));
+    });
+  }
+
+  private prepareUserTypePayload() {
+    this.userType.map(userType => {
+      this.userTypesPayload.push(this.staffFilterOptions.userTypes.find(userTypes => userTypes.key === userType));
+    });
+  }
+
+  private prepareLocationPayload() {
+    const locationPayload = [];
+    locationPayload.push({
+      location_id: this.primaryLocations.court_venue_id,
+      location: this.primaryLocations.site_name,
+      is_primary: true
+    });
+    this.additionalLocations.map(location => {
+      locationPayload.push({
+        location_id: location.court_venue_id,
+        location: location.site_name,
+        is_primary: false
+      });
+    });
+    return locationPayload;
+  }
+
+  private prepareRolesPayload() {
+    const roleObj = [
+      { key: 'case-allocator', label: 'Case Allocator' },
+      { key: 'task-supervisor', label: 'Task Supervisor' },
+      { key: 'staff-administrator', label: 'Staff Administrator' }
+    ];
+    this.roles.map(role => {
+      this.rolesPayload.push(roleObj.find(roles => roles.key === role));
+    })
+  }
+
+  private prepareRegionPayload() {
+    const regionObj = [
+      { key: 'region-1', label: 'Region 1'}
+    ];
+
+    this.region.map(region => {
+      this.regionPayload.push(regionObj.find(regions => regions.key === region));
+    })
+  }
+
+  private prepareSkillsPayload() {
+    const skillsPayload = [[]];
+    let nonEmptySkillsPayload = [[]];
+    this.skills.map(skill => {
+      this.staffFilterOptions.skills.map(skillgroup => {
+        skillsPayload.push(skillgroup.options.filter(skills => skills.key === skill));
+      });
+    });
+
+    nonEmptySkillsPayload = skillsPayload.filter(skill => skill.length > 0);
+
+    this.skillsPayload = nonEmptySkillsPayload.reduce((a, b) => {
+      return a.concat(b);
+    });
+  }
+
+  public addNewUser() {
+    let task_supervisor_flag = false;
+    let case_allocator_flag = false;
+    let staff_admin_flag = false;
+    this.roles.map(role => {
+      if (role === 'task-supervisor') {
+        task_supervisor_flag = true;
+        return;
+      }
+      if (role === 'case-allocator') {
+        case_allocator_flag = true;
+        return;
+      }
+      if (role === 'staff-administrator') {
+        staff_admin_flag = true;
+        return;
+      }
+    });
+
+    const addNewUserPayload = {
+      email_id: this.email,
+      first_name: this.firstName,
+      last_name: this.lastName,
+      services: this.servicePayload,
+      region: this.regionPayload[0].label,
```

Comment:
Can we have `if` condition to check the length of `this.regionPayload` as in case of zero length, I will throw a run time error.